modules/core/src/effects/mask/mask-effect.js

```diff
@@ -66,7 +66,12 @@ export default class MaskEffect extends Effect {
 
         channelInfo.bounds = getMaskBounds({layers: channelInfo.layers, viewport});
 
-        if (maskChanged || !equals(channelInfo.bounds, oldChannelInfo.bounds)) {
+        if (
+          maskChanged ||
+          !equals(channelInfo.bounds, oldChannelInfo.bounds) ||
+          !equals(channelInfo.coordinateOrigin, oldChannelInfo.coordinateOrigin) ||
```

Comment:
I don't think this is necessary. There is virtually no chance of a layer changing its coordinate system without changing its data.